src/traces/pie/plot.js

```diff
@@ -1089,7 +1092,7 @@ function setCoords(cd) {
         cdi.largeArc = (cdi.v > cd0.vTotal / 2) ? 1 : 0;
 
         cdi.halfangle = Math.PI * Math.min(cdi.v / cd0.vTotal, 0.5);
-        cdi.ring = 1 - trace.hole;
+        cdi.ring = 1 - (trace.hole[i] || trace.hole);
```

Comment:
Also here we should use `Lib.isArrayOrTypedArray` to test `trace.hole` before try accessing its index.

Reply:
👍